added exclude_pattern attribute to ignore certain files

davisfeng70 opened the pull request:
Added an extra parameter to dx-upload-play.yml to include an option to put an exclude pattern to exclude certain files from getting uploaded to DNAnexus.

The functionality already existed in dx_sync_directory.py (used to exclude 'Image'), and in incremental_upload.py.

Tested by including and not including the exclude_pattern attribute in dx-upload-play.yml.

davisfeng70 committed:
add exclude parameter

davisfeng70 committed:
added exclude_patterns definition

davisfeng70 committed:
re-add accidental line removal

orodeh reviewed (approved):
Looks good to me. I took a look at the python code, for the first time, and it looks well written.

davisfeng70 committed:
Update .travis.yml

force ubuntu 14.04 to resolve travis error

davisfeng70 committed:
corrected attribute name to exclude_patterns

Corrected attribute name from exclude_pattern to exclude_patterns to match documentation
